### PR TITLE
move text-input-v1 to hyprwayland-scanner

### PR DESCRIPTION
**Pull request by ikalco**
#### Describe your PR, what does it fix/add?
^

#### Is there anything you want to mention? (unchecked code, possible bugs, found problems, breaking compatibility, etc.)
tested with `chromium --enable-features=UseOzonePlatform --ozone-platform=wayland --enable-wayland-ime`
and works the same as before for me

#### Is it ready for merging, or does it need work?
yes

### CODE REVIEW

**Comment by ikalco** [comment]
im gonna do the last one soon so its possible to remove wayland-scanner from deps in that pr right?

File: src/protocols/TextInputV1.cpp [hunk 1]
```diff
@@ -3,228 +3,130 @@
...
+        pendingSurrounding.isPending = false;
+        pendingContentType.isPending = false;
+    });
 
-static void handleActivate(wl_client* client, wl_resource* resource, wl_resource* seat, wl_resource* surface) {
-    g_pProtocolManager->m_pTextInputV1ProtocolManager->handleActivate(client, resource, seat, surface);
-}
+    resource->setSetSurroundingText([this](CZwpTextInputV1* pMgr, const char* text, uint32_t cursor, uint32_t anchor) {
+        pendingSurrounding = {true, std::string(text), cursor, anchor};
+        ;
```
Comment (vaxerski) [inline src/protocols/TextInputV1.cpp:40]: e?

File: src/protocols/TextInputV1.cpp [hunk 2]
```diff
@@ -3,228 +3,130 @@
...
+        pendingContentType = {true, hint == (uint32_t)ZWP_TEXT_INPUT_V1_CONTENT_HINT_DEFAULT ? (uint32_t)ZWP_TEXT_INPUT_V1_CONTENT_HINT_NONE : hint,
+                              purpose > (uint32_t)ZWP_TEXT_INPUT_V1_CONTENT_PURPOSE_PASSWORD ? hint + 1 : hint};
+    });
 
-static void handleShowInputPanel(wl_client* client, wl_resource* resource) {
-    g_pProtocolManager->m_pTextInputV1ProtocolManager->handleShowInputPanel(client, resource);
-}
+    resource->setSetCursorRectangle([this](CZwpTextInputV1* pMgr, int32_t x, int32_t y, int32_t width, int32_t height) {
+        cursorRectangle = CBox{x, y, width, height};
+        ;
```
Comment (vaxerski) [inline src/protocols/TextInputV1.cpp:50]: ?

File: src/protocols/TextInputV1.cpp (hunk 1 above)
Comment (ikalco) [reply src/protocols/TextInputV1.cpp:40]: irs pretty lol

**Review by vaxerski — DISMISSED** [review]
mashallah brother

**Comment by vaxerski** [comment]
ay yo no pch fail lmao

**Comment by ikalco** [comment]
should be fixed now

**Pull request merged**